mavlink_vehicles/mavlink_vehicles.cc

```diff
@@ -739,6 +745,9 @@ void mav_vehicle::takeoff()
 
 void mav_vehicle::rotate(double angle_deg, bool autocontinue)
 {
+    // We have taken control
+    this->is_our_control = true;
```

Comment:
The drone is not exactly aware of whom is controlling it. It can be controlled by anyone that is connected to it. What we are doing here is only relevant to the mavlink_vehicles layer - to prevent multiple instances to try to autorotate/autocontinue during a detour or mission.

Every time a instance of a mavlink_vehicle sends a mission command to the vehicle, arducopter broadcasts a mission ACK saying that it has succesfuly received the command. This mission ACK is then parsed by every mavlink_vehicle instance connected to arducopter. If that ack is not being expected - i.e., the instance has not sent any mission command whose ack has not been received yet - that instance knows that someone else is trying to control the vehicle, so it stops to try to autorotate/autocontinue until either one of its command functions - rotate(), brake(), send_waypoint(), ... - is called.

In other words, the responsibility to autorotate/autocontinue is always in the hands of the last mavlink_vehicle instance that has sent a mission command.